metric/instrument/asyncint64/asyncint64.go

```diff
@@ -14,53 +14,28 @@
 
 package asyncint64 // import "go.opentelemetry.io/otel/metric/instrument/asyncint64"
 
-import (
-	"context"
+import "go.opentelemetry.io/otel/metric/instrument"
 
-	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/metric/instrument"
-)
-
-// Counter is an instrument that records increasing values.
+// Counter is an instrument used to asynchronously record increasing int64
+// measurements once per a measurement collection cycle. The Observe method is
+// used to record the measured state of the instrument when it is called.
+// Implementations will assume the observed value to be the cumulative sum of
+// the count.
 //
 // Warning: methods may be added to this interface in minor releases.
-type Counter interface {
-	// Observe records the state of the instrument to be x. Implementations
-	// will assume x to be the cumulative sum of the count.
-	//
-	// It is only valid to call this within a callback. If called outside of the
-	// registered callback it should have no effect on the instrument, and an
-	// error will be reported via the error handler.
-	Observe(ctx context.Context, x int64, attrs ...attribute.KeyValue)
-
-	instrument.Asynchronous
-}
+type Counter interface{ instrument.Int64Observer }
```

Comment:
Is there a reason why these are not defined the same way as the Sync counterparts?

Reply:
Just that we needed the `Int64Observer` and `Float64Observer` types explicitly to reference in their respective callbacks. And once that was added it made sense to simplify this definition. I think we can do the same for the sync instruments, but I'm not sure what name we would give to the type.

I will plan to explore this in the follow up PR planned to flatten the async/sync pacakges.